apps/wallet/src/background/keyring/vault/index.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { encrypt, decrypt } from '_shared/cryptography/keystore';
+import {
+    mnemonicToEntropy,
+    validateMnemonics,
+    entropyToMnemonic,
+} from '_shared/utils/bip39';
+
+export type StoredData = string | { v: number; data: string };
+
+/**
+ * Holds the mnemonic of the wallet and provides functionality to create/encrypt/decrypt it.
+ */
+export class Vault {
+    readonly #version = 1;
+    public readonly mnemonic: string;
```

Comment:
We shouldn't store this, we should only derive it when it is requested.

Reply:
You mean not keep it in memory right? Since we are not storing to disk

Reply:
Yeah exactly, we should try to not keep the mnemonic held in memory longer than the function that needs it (i.e. derive mnemonic, use it, unreference).